templates/item/annotation.macros.j2

```diff
@@ -0,0 +1,297 @@
+{% macro requires(item, display_private=false) -%}
+  {% if (item.require|length > 0) %}
+    {% call item_section('requires') %}
+      {% set prev = '' %}
+
+      {% for req in item.require %}
+        {% set name = req.name if req.name else req.item.context.name %}
+        {% set type = req.type if req.type else (req.item.context.type or none) %}
+        {% set check = [name|string(), type|string()]|join('') %}
+
+        {% if not (check in prev) %}
+          {% set prev = [prev, ', ', check]|join if (prev != '') else check %}
+          {{ require_item(req, display_private) }}
+        {% endif %}
+      {% endfor %}
+    {% endcall %}
+  {% endif %}
+{%- endmacro %}
+
+
+{# @require items #}
+{% macro require_item(req, display_private=false) %}
+  {% set require_name = show.format_name(req.name, req.type) %}
+  {% set require_type = show.format_type(req.type) %}
+  {% set is_hidden = true if (req.item.access == 'private' and not display_private) else false %}
+  {% set require_url = none if is_hidden else show.get_link(req.name, req.type, req.item.group) %}
+  {% set require_url = req.url if req.url else require_url %}
+  {% set value_type = req.item.type if (req.type == 'variable') else none %}
+  {% set note = 'private' if (req.item.access == 'private') else none %}
+  {% set note = 'external' if req.external else note %}
+
+  {{ param_list(
+    name=require_name,
+    url=require_url,
+    item_type=require_type,
+    value_type=value_type,
+    note=note,
+    description=req.description
+    ) }}
+{% endmacro %}
+
+
+{# @link and @see annotations #}
+{% macro related(item, display_private=false) -%}
+  {% if (item.link.length > 0) or (item.see.length > 0) %}
+
+    {% call item_section('related') %}
+      {% for link in item.link %}
+        {{ link_item(link) }}
+      {% endfor %}
+
+      {% for see in item.see %}
+        {{ see_item(see, display_private) }}
+      {% endfor %}
+    {% endcall %}
+  {% endif %}
+{%- endmacro %}
+
+
+{# @link items #}
+{% macro link_item(link) %}
+  {% set note = 'external' if ('://' in link.url) else none %}
+
+  {{ param_list(
+    name=link.caption or link.url,
+    url=link.url,
+    note=note
+    ) }}
+{% endmacro %}
+
+
+{# @see items #}
+{% macro see_item(item, display_private=false) %}
+  {% set private = (item.access == 'private') and (not display_private) %}
+  {% set url = show.get_link(item.context.name, item.context.type, item.group) if not private else none %}
+  {% set name = show.format_name(item.context.name, item.context.type) %}
+  {% set type = show.format_type(item.context.type) %}
+  {% set note = 'private' if private else none %}
+
+  {{ param_list(
+    name=name,
+    url=url,
+    item_type=type,
+    value_type=value_type,
+    note=note,
+    description=item.context.description
+    ) }}
+{% endmacro %}
+
+
+{# @todo annotation #}
+{% macro todo(item) %}
+  {% if (item.todo.length > 0) %}
+    {% call item_section('todo', 'Todo List') %}
```

Comment:
This sent @SondraE down a rabbit-hole of googling. A google-hole? Seems the "proper" way, according to various dictionaries, would be "to-do list". :)